pandas/io/packers.py

```diff
@@ -595,11 +595,14 @@ def decode(obj):
         pd_dtype = pandas_dtype(dtype)
         np_dtype = pandas_dtype(dtype).base
 
+        # see gh-14901
+        klass_dtype = np_dtype if pd_dtype.name != 'category' else pd_dtype
+
         index = obj[u'index']
         result = globals()[obj[u'klass']](unconvert(obj[u'data'], dtype,
                                                     obj[u'compress']),
                                           index=index,
-                                          dtype=np_dtype,
+                                          dtype=klass_dtype,
```

Comment:
@jreback : It is absolutely relevant.  The test for `Series` equality is based on these checks.  Here is  an easy way to reproduce (similar to what happens internally in `read_msgpack`):
~~~python
>>> from pandas import Series, Timestamp
>>> s = Series([Timestamp('20130102', tz='US/Eastern')] * 5)
>>> s2 = Series(s.get_values(), dtype=s.dtype)
>>>
>>> s.get_values()
array(['2013-01-02T05:00:00.000000000', '2013-01-02T05:00:00.000000000',
       '2013-01-02T05:00:00.000000000', '2013-01-02T05:00:00.000000000',
       '2013-01-02T05:00:00.000000000'], dtype='datetime64[ns]')
>>>
>>> s2.get_values()
array(['2013-01-02T10:00:00.000000000', '2013-01-02T10:00:00.000000000',
       '2013-01-02T10:00:00.000000000', '2013-01-02T10:00:00.000000000',
       '2013-01-02T10:00:00.000000000'], dtype='datetime64[ns]')
~~~